### PR TITLE
Adds pagination to repositories index

### PR DESCRIPTION
**Pull request by marcosddourado**
Adds will_paginate gem
Adds functional tests of pagination
Adds step definition to click on link based on its text
Adds will_paginate initializer to workaround repositories not being child of activerecord::base

### CODE REVIEW

**Comment by DylanGuedes** [comment]
This merge closes the issue #361

**Comment by rafamanzo** [comment]
Nicely done. Please rebase from master. This will require further improvement once the necessary API gets added into KalbroProcessor and KalibroClient (so the requests also can get paginated) as I've tlaked to @DylanGuedes last week on IRC.

File: features/repository/index.feature [hunk 1]
```diff
@@ -35,3 +35,24 @@ Feature: Repository listing
...
+    And I have a public repository named "reposittory 4"
+    And I have a public repository named "reposittory 5"
+    And I have a public repository named "reposittory 6"
+    And I have a public repository named "reposittory 7"
+    And I have a public repository named "reposittory 8"
+    And I have a public repository named "reposittory 9"
+    And I have a public repository named "reposittory 10"
+    And I own that repository
+    And I am at the All Repositories page
+    Then I should see a div with class "pagination"
```
Comment (rafamanzo) [inline features/repository/index.feature]: What do you think about expanding this in order to check if the correct repositories are at the correct pages? This implies in testing a second page as well.

File: features/step_definitions/repository_steps.rb [hunk 2]
```diff
@@ -295,6 +295,10 @@
   expect(page.driver.status_code).to eq(404)
 end
 
+Then(/^I should see a div with class "(.*?)"$/) do |arg1|
+  page.all("div.#{arg1}")
```
Comment (rafamanzo) [inline features/step_definitions/repository_steps.rb]: Does `page.all` fail if the element has not been found or it just returns an empty array?

In the lines just below you can a similar selector that we know for sure to fail when is necessary: `expect(page).to have_selector`

**Comment by diegoamc** [comment]
ping @marcosddourado, @DylanGuedes

File: app/controllers/repositories_controller.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ class RepositoriesController < ApplicationController
 
   def index
     @repositories = Repository.public_or_owned_by_user(current_user)
+
+    @repositories = @repositories.paginate(:page => params[:page], :per_page => 10)
```
Comment (diegoamc) [inline app/controllers/repositories_controller.rb]: Where does the `:page` param come from?

**Comment by vitorbaraujo** [comment]
What is the current status of this PR?

**Comment by DylanGuedes** [comment]
@vitorbaraujo [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/mezuro/LZAsVc51FbI). We solved it in the naive way, so you can check how we did it for educational purposes, but the correct solution shouldn't use our solution.

**Comment by rafamanzo** [comment]
This PR does not reduce the size of the data downloaded in a single request. If you want to keep working on this issue, please submit another PR.

**Comment by paulormm** [comment]
@vitorbaraujo , would you talk to @dylanguedes to the MES'2017 team proceed a new PR based on this PR?

**Pull request closed without merging**